src/vis/vis.js

```diff
@@ -424,14 +423,26 @@ var Vis = View.extend({
     });
 
     // Trigger 'done' event
-
     _.defer(function () {
       self.trigger('done', self, self.map.layers);
     });
 
+    if (!options.skipMapInstantiation) {
+      this.instantiateMap();
+    }
+
     return this;
   },
 
+  instantiateMap: function (invalidateMapSize) {
+    this._dataviewsCollection.on('add reset remove', _.debounce(this._invalidateSizeOnDataviewsChanges, 10), this);
+    this.map.instantiateMap();
+
+    if (invalidateMapSize) {
```

Comment:
Only invalidate map size when it is requested. In CartoDB.js visualizations is not needed, because the map size doesn't change. For DI is a different story.